### DISCUSSION TITLE
Ability to use this package outside of filament admin panel

### DISCUSSION THREAD

**Discussion by CovertError** [ysfkaya/filament-phone-input#10] [Ideas]
I keep getting errors in my console when I try to use it outside of filament

**Comment by ysfkaya** [comment]
I have plans for this package to work outside the filament admin in version 2. Therefore, you need to integrate the asset files that the package uses into the system yourself. The filament admin does this. I think it will be enough to implement only asset files in your system.

**Comment by Saifallak** [comment]
any idea how to do that?

**Reply by ysfkaya** [reply]
In the real life, I needed the phone input in some cases. So I just move the `js` builds to `app.js`.

Here is an example to usage it outside from Filament:

`app.js`

```js
import "./bootstrap";

import Alpine from "alpinejs";

import "../../vendor/ysfkaya/filament-phone-input/dist/intl-tel-input/utils";
import "../../vendor/ysfkaya/filament-phone-input/dist/js/filament-phone-input";
import "../../vendor/ysfkaya/filament-phone-input/dist/css/intl-tel-input.css";
import "../../vendor/ysfkaya/filament-phone-input/dist/css/filament-phone-input.css";

Alpine.plugin(Alert);

window.Alpine = Alpine;

Alpine.start();
```

`input.blade.php`

```blade
   <div wire:ignore.self class="w-full">
      <input type="tel" @class([
          '...',
          'border border-monza-500' => $errors->has('form.phone'),
      ]) x-data="phoneInputFormComponent({
          getInputTelOptionsUsing: () => ({
              geoIpLookup: 'ipinfo',
              initialCountry: 'auto',
              inputNumberFormat: 'E164',
              displayNumberFormat: 'NATIONAL'
          }),
          state: $wire.entangle('form.phone').defer,
      })">
  </div>
```

As you see I don't use the `form` feature of Filament yet. 

I'm thinking new way to usage with `form` feature it in the next version of this package. Maybe just we can add assets on the top of application's layout. Just like below

`components/layouts/app.blade.php`

```blade

<body>
   ....
   
   @phoneInputScripts
</body>
```

That's it. Maybe in `v3` of Filament this may not be needed. I'll look into this once Filament v3 goes live.

**Comment by ysfkaya** [comment]
You can find an example about it [here](https://github.com/ysfkaya/filament-phone-input#outside-filament) in `v2`
